internal/daemon/device-handler/sriov-device-handler/sriovdevicehandler.go

```diff
@@ -75,11 +107,50 @@ func (nf *sriovDeviceHandler) GetDevices() (*dp.DeviceList, error) {
 	return &devices, nil
 }
 
+// ensureConnected makes sure we are connected to the VSP's gRPC
+func (s *sriovDeviceHandler) ensureConnected() error {
+	// TODO: FIXME, design proto API for VSP
+	return nil
+}
+
+// SetupDevices
+func (s *sriovDeviceHandler) SetupDevices() error {
+	s.setupDevicesDone = make(chan struct{})
+
+	err := s.ensureConnected()
+	if err != nil {
+		return fmt.Errorf("failed to ensure connection to vsp: %v", err)
+	}
+
+	// TODO: The VSP should pass the hardcoded parameters to create a filter
+	s.vfFilterFunc = CreatePcieDevFilter("8086", "145c", "idpf")
+
+	// TODO: The VSP should pass in the PF
+	err = s.SetSriovNumVfs("0000:06:00.0", 8)
+	if err != nil {
+		return fmt.Errorf("failed to set sriov numVfs: %v", err)
+	}
+
+	close(s.setupDevicesDone)
+
+	return nil
+}
+
 func NewSriovDeviceHandler() *sriovDeviceHandler {
-	// TODO: The VSP should pass the following parameters to create a filter
-	filterFunc := CreatePcieDevFilter("8086", "145c", "idpf")
-	return &sriovDeviceHandler{
-		log:        ctrl.Log.WithName("SriovDeviceHandler"),
-		filterFunc: filterFunc,
+	devHandler := &sriovDeviceHandler{
+		log: ctrl.Log.WithName("SriovDeviceHandler"),
 	}
+
+	// TODO: When changing the SRIOV numVfs, we should do the following:
+	// 1) Drain all pods running on the node with a drain controller running
+	// on the control plane. The nodes will be marked for draining and read by
+	// the drain controller.
+	// 2) Clearly define which PF and how many VFs from an API. User facing or
+	// otherwise
+	err := devHandler.SetupDevices()
```

Comment:
Since we call SetupDevice here, it will call EnsureConnected. This introduces an order: daemon needs to be started after VSP, which we want to avoid.

Currently, you are printing an error but this means that the daemon goes into a broken state that we will need to recover by starting VSP and then restarting the daemon pod.

Reply:
Should we block for the VSP here? There is no integration with VSP yet, but I was wondering if this will be an issue if the VSP is part of the same pod as the daemon.